### PR TITLE
06 task - 

### PR DESCRIPTION
**Pull request by mollybrace**
Tasks completed up to ticket 7  vote on article.

### CODE REVIEW

File: src/components/comments.jsx [hunk 1]
```diff
@@ -1,8 +1,40 @@
...
+const Comments = ({article_id}) => {
+
+const [comments, setComments] = useState([])
+    
+useEffect (() => {
+    fetchComments(article_id).then((commentsReceived) => {
+        setComments(commentsReceived)
+    })
+}, [article_id])
+
```
Comment (Jodiech) [inline src/components/comments.jsx:18]: Consider adding visual feedback if data is still loading for the user 👍

File: src/components/articleelement.jsx [hunk 2]
```diff
@@ -26,6 +34,8 @@ return (
                             <h3>{info.title}</h3>
                             <img src={info.article_img_url} alt="Article image"></img>
                             <p>{info.body}</p>
+                            <p>Votes: {info.votes}</p>
+                            <button onClick={handleClick}>Vote</button>
                             <p>{info.author}</p>
                             <p>{info.created_at}</p>
                         </li>
```
Comment (Jodiech) [inline src/components/articleelement.jsx:67]: Nice to see the vote updating immediately with optimistic rendering 👍 You could also add a downvote as well. Just a suggestion but something to think about for your app 😄

File: src/components/articleelement.jsx [hunk 3]
```diff
@@ -13,7 +16,12 @@ useEffect(() => {
         setArticle(articleReceived)
     })
 })
-console.log(article)
+
+const handleClick = () =>{
+    patchArticleVotes(article_id)
```
Comment (Jodiech) [inline src/components/articleelement.jsx]: Consider adding a message for the user if the patch request fails.

**Review by Jodiech — CHANGES_REQUESTED** [review]
Hey Molly 😄 well done on the tickets! For ticket 7 I'd recommend adding functionality so the voter can only vote once in either direction. I've left a comment for adding visual feedback for the user if the data is loading. I'd add a message for the user if the patch request fails. Well done! 🚀

**Comment by Harrison365** [comment]
Hey Molly :) 

This is looking good 👍 

Please could you take a look at the following:
- ensure that people can't post an empty comment (optimistically). E.g. give an error message.
- forms should be cleared after submit.
- user should be given feedback after a successful or unsuccessful post 
- the comment section should be multi-line (e.g. text-area not input)

Please let us know when you'd like us to take another look 🙂

**Comment by robert112233** [comment]
Your search by topic is working great! You might want to add an isLoading in between the searches to show the user that something is taking place under the hood. 

I'm not sure if you needed comment submission reviewing but they aren't working as expected still. When I try to submit a blank comment I get "comment submitted successfully" _and_ an error message. And then when I do submit a valid comment, the error message from before still persists. I've attached screenshots to show this behaviour. 

Screenshot 2023-03-16 at 13 50 06
Screenshot 2023-03-16 at 13 51 53

File: src/components/articleelement.jsx (hunk 2 above)
Comment (mollybrace) [reply src/components/articleelement.jsx:67]: completed

Comment (Jodiech) [reply src/components/articleelement.jsx:67]: Hey Molly, Great that you have put in the functionality for upvoting and downvoting. I'd change these to buttons as it's unclear to see if they are two separate things. You could disable one of the buttons when one is clicked to show the user they have pressed it. Hope this makes sense, let me know if you need help with anything 😄

**Pull request merged**